dashboard/src/components/DeploymentForm/DeploymentFormBody/BasicDeploymentForm/TabularSchemaEditorTable/Params/ArrayParam.tsx

```diff
@@ -0,0 +1,166 @@
+// Copyright 2022 the Kubeapps contributors.
+// SPDX-License-Identifier: Apache-2.0
+
+import { CdsButton } from "@cds/react/button";
+import { CdsIcon } from "@cds/react/icon";
+import { CdsInput } from "@cds/react/input";
+import { CdsRange } from "@cds/react/range";
+import { CdsToggle } from "@cds/react/toggle";
+import Column from "components/js/Column";
+import Row from "components/js/Row";
+import { useState } from "react";
+import { IBasicFormParam } from "shared/types";
+import { basicFormsDebounceTime } from "shared/utils";
+
+export interface IArrayParamProps {
+  id: string;
+  label: string;
+  type: string;
+  param: IBasicFormParam;
+  handleBasicFormParamChange: (
+    param: IBasicFormParam,
+  ) => (e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>) => void;
+}
+
+export default function ArrayParam(props: IArrayParamProps) {
+  const { id, label, type, param, handleBasicFormParamChange } = props;
+
+  const [currentArrayItems, setCurrentArrayItems] = useState<(string | number | boolean)[]>(
+    param.currentValue ? JSON.parse(param.currentValue) : [],
+  );
+  const [timeout, setThisTimeout] = useState({} as NodeJS.Timeout);
+
+  const setArrayChangesInParam = () => {
+    clearTimeout(timeout);
+    const func = handleBasicFormParamChange(param);
+    // The reference to target get lost, so we need to keep a copy
+    const targetCopy = {
+      currentTarget: {
+        value: JSON.stringify(currentArrayItems),
+        type: "change",
+      },
+    } as React.FormEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>;
+    setThisTimeout(setTimeout(() => func(targetCopy), basicFormsDebounceTime));
+  };
+
+  const onChangeArrayItem = (index: number, value: string | number | boolean) => {
+    currentArrayItems[index] = value;
+    setCurrentArrayItems([...currentArrayItems]);
+    setArrayChangesInParam();
+  };
+
+  const renderInput = (type: string, index: number) => {
+    switch (type) {
+      case "number":
+      case "integer":
+        return (
+          <>
+            <CdsInput className="self-center">
+              <input
+                required={param.required}
```

Comment:
Adding this basic HTML-based validation for solving this issue: https://github.com/vmware-tanzu/kubeapps/issues/1924